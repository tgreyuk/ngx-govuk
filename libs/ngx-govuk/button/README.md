# ngx-govuk/button

Secondary entry point of `ngx-govuk`. It can be used by importing from `ngx-govuk/button`.
