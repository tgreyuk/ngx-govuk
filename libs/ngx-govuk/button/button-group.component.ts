import { CommonModule } from '@angular/common';
import { ChangeDetectionStrategy, Component, HostBinding } from '@angular/core';
/**
 * Component that wraps a group of buttons. The content slot expects x number of buttons.
 *
 * @selector ngx-govuk-button-group
 */
@Component({
  selector: 'ngx-govuk-button-group',
  standalone: true,
  imports: [CommonModule],
  templateUrl: './button-group.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class NgxGovukButtonGroup {
  @HostBinding('class') get className() {
    return 'govuk-button-group';
  }
}
