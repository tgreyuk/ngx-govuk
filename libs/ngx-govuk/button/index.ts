export * from './button-group.component';
export * from './button.directive';
