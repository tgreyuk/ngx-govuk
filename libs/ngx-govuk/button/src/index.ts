export * from './button-group.directive';
export * from './button.directive';
