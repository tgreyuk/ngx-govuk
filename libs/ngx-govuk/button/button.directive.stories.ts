import { moduleMetadata, type Meta } from '@storybook/angular';
import { NgxGovukButtonGroup } from './button-group.component';
import { NgxGovukButton } from './button.directive';

const meta: Meta<NgxGovukButton> = {
  component: NgxGovukButton,
  title: 'UI Components/Button',
  decorators: [
    moduleMetadata({
      imports: [NgxGovukButtonGroup],
    }),
  ],
};
export default meta;

export const Button = {
  argTypes: {
    buttonType: {
      options: ['Default', 'secondary', 'warning', 'start'],
      control: { type: 'select' },
    },
    inversed: { type: 'boolean' },
  },
  render: (args: NgxGovukButton) => ({
    props: args,
    template: `
      <button ngxGovukButton [buttonType]="buttonType" [inversed]="inversed">Action</button>
    `,
  }),
};

export const ButtonGroup = {
  render: () => ({
    template: `
      <ngx-govuk-button-group>
        <button ngxGovukButton>Continue</button>
        <a href="#" ngxGovukButton buttonType="secondary">Cancel</a>
      </ngx-govuk-button-group>
    `,
  }),
};
