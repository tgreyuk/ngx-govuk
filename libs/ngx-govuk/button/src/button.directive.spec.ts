import { NgxGovukButtonDirective } from './button.directive';

describe('ButtonDirective', () => {
  it('should create an instance', () => {
    const directive = new NgxGovukButtonDirective();
    expect(directive).toBeTruthy();
  });
});
