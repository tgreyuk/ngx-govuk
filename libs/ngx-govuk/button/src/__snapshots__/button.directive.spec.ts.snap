// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NgxGovukButton should check markup snapshot 1`] = `
<ng-component>
  <button
    class="govuk-button"
    ngxgovukbutton=""
  >
    Default
  </button><button
    buttontype="warning"
    class="govuk-button govuk-button--warning"
    ngxgovukbutton=""
  >
    Warning
  </button><button
    buttontype="start"
    class="govuk-button govuk-button--start"
    ngxgovukbutton=""
  >
    Start
    <svg
      aria-hidden="true"
      class="govuk-button__start-icon"
      focusable="false"
      height="19"
      viewBox="0 0 33 40"
      width="17.5"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h13l20 20-20 20H0l20-20z"
        fill="currentColor"
      />
    </svg>
  </button>
</ng-component>
`;
